kong/runloop/plugins_iterator.lua

```diff
@@ -45,16 +45,26 @@ local function load_plugin_configuration(ctx,
     return ngx.exit(ngx.ERROR)
   end
 
-  if plugin ~= nil and plugin.enabled then
-    local cfg = plugin.config or {}
-
-    cfg.api_id      = plugin.api and plugin.api.id
-    cfg.route_id    = plugin.route and plugin.route.id
-    cfg.service_id  = plugin.service and plugin.service.id
-    cfg.consumer_id = plugin.consumer and plugin.consumer.id
+  if not plugin or not plugin.enabled then
+    return
+  end
 
-    return cfg
+  if plugin.run_on ~= "all" then
+    if ctx.is_service_mesh_request and plugin.run_on == "first" then
+      return
+    elseif plugin.run_on == "second" then
+      return
```

Comment:
Seems like plugins with `run_on == "second"` would never run, since they'd enter this branch every time.